client/client_test.go

```diff
@@ -1433,4 +1433,35 @@ var _ = Describe("Test the client query API", Ordered, func() {
 		Expect(gvr.Namespaced).To(BeTrue())
 		Expect(gvr.Resource).To(Equal("configmaps"))
 	})
+
+	It("A cluster scoped Get query with a namespace is handled correctly", func() {
```

Comment:
How hard would it be to add more tests to verify that updates on cluster-scoped resources are processed correctly? Like the `Gets an updated cache entry from an update` test earlier in this file. What I'd want to check for in particular is that setting a namespace on a cluster-scoped resource doesn't prevent the user from seeing the updates on it. I think that the code handles this, but a test would be extra helpful.

If the test setup is too cumbersome, I think it would be ok to merge as-is.

Reply:
@JustinKuli I added some more tests. Could you please take another look?